index.d.ts

```diff
@@ -344,6 +345,11 @@ export interface TimeoutFn {
 	(ms: number): void;
 }
 
+export interface TeardownFn {
+	/** Declare a hook that is run after the test, regardless if it passed or failed. */
+	(): ImplementationResult;
```

Comment:
This should take the hook as an argument, and return `void`.

Reply:
Went with `(hook: () => PromiseLike<any> | void): void`